### PR TITLE
nvfancontrol: init at 0.5.1

### PR DESCRIPTION
**Pull request by devins2518**
###### Things done
Added nvfancontrol

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/nvfancontrol/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  nativeBuildInputs = [ libXNVCtrl libX11 libXext ];
+
+  # Needed for static linking
+  preConfigure = ''
+    export LIBRARY_PATH=${libXNVCtrl}/lib:${libX11}/lib:${libXext}/lib
+  '';
+
+
+  meta = with lib; {
+    description = "NVidia dynamic fan control for Linux and Windows";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nvfancontrol/default.nix:25]:
```suggestion
    description = "NVidia dynamic fan control for Linux";
```

File: pkgs/tools/misc/nvfancontrol/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub, libXNVCtrl, libX11, libXext }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "nvfancontrol";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "foucault";
+    repo = pname;
+    rev = "${version}";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nvfancontrol/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/tools/misc/nvfancontrol/default.nix (hunk 1 above)
Comment (devins2518) [reply pkgs/tools/misc/nvfancontrol/default.nix]: Both fixed in eb4fdc9.

File: pkgs/tools/misc/nvfancontrol/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    export LIBRARY_PATH=${libXNVCtrl}/lib:${libX11}/lib:${libXext}/lib
+  '';
+
+
+  meta = with lib; {
+    description = "NVidia dynamic fan control for Linux";
+    homepage = "https://github.com/foucault/nvfancontrol";
+    changelog = "https://github.com/foucault/nvfancontrol/releases/tag/${version}";
+    license = with licenses; [ gpl3Only ];
+    maintainers = with maintainers; [ devins2518 ];
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nvfancontrol/default.nix:29]:
```suggestion
    platforms = platforms.linux;
    maintainers = with maintainers; [ devins2518 ];
```

File: pkgs/tools/misc/nvfancontrol/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub, libXNVCtrl, libX11, libXext }:
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nvfancontrol/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, libXNVCtrl, libX11, libXext }:
```
i don't think it's needed

Comment (devins2518) [reply pkgs/tools/misc/nvfancontrol/default.nix]: Yeah, forgot to remove it. Fixed in 045adfc.

File: pkgs/tools/misc/nvfancontrol/default.nix (hunk 3 above)
Comment (devins2518) [reply pkgs/tools/misc/nvfancontrol/default.nix:30]: Fixed in 045adfc.

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 125070` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nvfancontrol</li>
  </ul>
</details>
It builds, but errors with:
```
WARN - No config file found; using default curve
ERROR - Could not get driver version: XNVCTRLIsNvScreen failed; no screens assigned to the NVidia driver
```
And yes, I have a NVidia card.

**Comment by devins2518** [comment]
Strange, do you have a dual GPU setup?

**Comment by legendofmiracles** [comment]
I do, but it's a probably upstream error.

**Comment by devins2518** [comment]
Yup, it seems that it only works if the X session is attached to the card. [The creator did link a way to possibly make this work through faking a display.](https://github.com/foucault/nvfancontrol/issues/11)

**Review by legendofmiracles — APPROVED** [review]

**Comment by teto** [comment]
Just as a datapoint I get:
```
sudo results/nvfancontrol/bin/nvfancontrol 
[sudo] password for teto: 
WARN - No config file found; using default curve
INFO - NVIDIA driver version: 460.73.01
INFO - NVIDIA graphics adapter #0: GeForce GTX 970
INFO -   GPU #0 coolers: COOLER-0
ERROR - Could not update fan speed: XNVCtrl QueryAttr(COOLER_MANUAL_CONTROL) failed; error 0
ERROR - Could not update fan speed: XNVCtrl QueryAttr(COOLER_MANUAL_CONTROL) failed; error 0
```

**Comment by devins2518** [comment]
You need coolbits enabled in your xorg.conf. [My dotfiles have an example here.](https://github.com/devins2518/dotfiles/blob/8697016e242195476a5e788b86d697e124aef139/hosts/pain/configuration.nix#L51)

File: pkgs/tools/misc/nvfancontrol/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ libXNVCtrl libX11 libXext ];
+
+  # Needed for static linking
+  preConfigure = ''
+    export LIBRARY_PATH=${libXNVCtrl}/lib:${libX11}/lib:${libXext}/lib
+  '';
+
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/nvfancontrol/default.nix:24]:
```suggestion
  '';

  meta = with lib; {
```

**Pull request merged**